library/bignum_core.h

```diff
@@ -398,6 +398,25 @@ mbedtls_mpi_uint mbedtls_mpi_core_mla(mbedtls_mpi_uint *X, size_t X_limbs,
                                       const mbedtls_mpi_uint *A, size_t A_limbs,
                                       mbedtls_mpi_uint b);
 
+/**
+ * \brief Perform a known-size multiplication
+ *
+ * \p A may be aliased to \p B.
```

Comment:
By default we promise that the output may be aliased to one or more inputs: https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.4.0/library/bignum_core.h#L44-L47.

We should explicitly state that X may not be aliased to any of the inputs for this function.